docs/Architecture_Overview.md

```diff
@@ -0,0 +1,85 @@
+# Architecture Overview
+
+This document provides an overview of the system architecture, explaining how different components interact with each
+other.
+
+## High-Level Components
+
+The system is composed of several high-level components:
+
+1. **Hyperlane Smart Contracts**: The core smart contracts that facilitate cross-chain communication.
+2. **Mailbox**: A central component for relaying messages between chains.
+3. **Relayer**: An off-chain service that listens to events on one chain and relays them to another chain.
+4. **Interchain Security Module (ISM)**: Ensures the security of cross-chain transactions by validating message
+   signatures.
+
+## Hyperlane Smart Contracts
+
+The Hyperlane smart contracts are designed to facilitate cross-chain communication. They include:
+
+- `HypERC20`: An ERC20 token with remote transfer functionality.
```

Comment:
Would need to add two extra bullet points here for `HypERC20Collateral` and `HypLSP7Collateral`